WireAPI/Sources/WireAPI/Models/Backend/PinnedKey.swift

```diff
@@ -0,0 +1,74 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+@preconcurrency import Security
+
+/// Associates a list of `hosts` with a public `key`.
+
+public struct PinnedKey: Sendable {
+
+    public enum Failure: Error {
+        case invalidKeyData
+    }
+
+    public enum Host: Sendable {
+        case endsWith(String)
+        case equals(String)
+    }
+
+    let key: SecKey
+    let hosts: [Host]
+
+    public init(key: SecKey, hosts: [Host]) {
```

Comment:
This is a temporary method for gluing things together under pressure of release. I need to think a bit more about how and where to map the data from `backend.bundle` but expect this method to be deleted in a follow up PR and instead favor throwing version.